File: src/test/java/destiny/data/gauquelin/weka/house_aspect/GPersonInstanceTest.java
/**
 * Created by smallufo at 2009/3/19 下午 7:50:54
 */
package destiny.data.gauquelin.weka.house_aspect;

import weka.classifiers.Classifier;
import weka.core.Instance;
import destiny.data.gauquelin.GDao;
import destiny.data.gauquelin.GPerson;

//TODO : JUnit4
public class GPersonInstanceTest
{
  protected GDao gDao;

  public void testGetInstance() throws Exception
  {
    GPerson gperson = gDao.get(8943L).get();
    System.out.println("gperson = " + gperson);
    GPersonInstance inst = new GPersonInstance(gperson);
    Instance instance = inst.getInstance();
    System.out.println(instance);
    
    Classifier c = new WekaWrapper();
    double[] fDistribution = c.distributionForInstance(instance);
    for (double d : fDistribution)
      System.out.println("d = " + d);
  }

}

File: src/test/java/destiny/data/gauquelin/weka/house_aspect/GPersonInstanceTest.kt
/**
 * Created by smallufo at 2009/3/19 下午 7:50:54
 */
package destiny.data.gauquelin.weka.house_aspect

import destiny.data.gauquelin.GDao
import destiny.data.gauquelin.GPerson
import destiny.tools.spring.get
import weka.classifiers.Classifier

//TODO : JUnit4
class GPersonInstanceTest {

  protected lateinit var gDao: GDao

  @Throws(Exception::class)
  fun testGetInstance() {
    val gperson: GPerson = gDao.get(8943L)!!
    println("gPerson = $gperson")
    val inst = GPersonInstance(gperson)
    val instance = inst.instance
    println(instance)
    val c: Classifier = WekaWrapper()
    val fDistribution = c.distributionForInstance(instance)
    for (d in fDistribution) println("d = $d")
  }
}
